src/Osiset/BasicShopifyAPI/BasicShopifyAPI.php

```diff
@@ -120,12 +120,20 @@ public function __construct(
 
         // Create the stack and assign the middleware which attempts to fix redirects
         $this->stack = HandlerStack::create($this->getOptions()->getGuzzleHandler());
-        $this
-            ->addMiddleware(new AuthRequest($this), 'request:auth')
-            ->addMiddleware(new RateLimiting($this), 'rate:limiting')
-            ->addMiddleware(new UpdateApiLimits($this), 'rate:update')
-            ->addMiddleware(new UpdateRequestTime($this), 'time:update')
-            ->addMiddleware(GuzzleRetryMiddleware::factory(), 'request:retry');
+        if ($this->getOptions()->getUseRateLimit()) {
```

Comment:
@ortegafernando Better to leave it as it was, then optionally add in the rate limiting... example:

```php
            $this
                ->addMiddleware(new AuthRequest($this), 'request:auth')
                ->addMiddleware(new UpdateApiLimits($this), 'rate:update')
                ->addMiddleware(new UpdateRequestTime($this), 'time:update')
                ->addMiddleware(GuzzleRetryMiddleware::factory(), 'request:retry');
         if ($this->getOptions()->getUseRateLimit()) {
              $this->addMiddleware(new RateLimiting($this), 'rate:limiting')
         }
```